### PR TITLE
Mobile Manipulator Bug Fix

### PR DESCRIPTION
**Pull request by ASzot**
## Motivation and Context

This PR fixes two issues:
1. The mobile manipulator joint creation was incorrect. The arm joint parameters were being applied to all joints rather than just the arm joints. The default motor parameters should be applied to the non-arm and non-base parameters. Fixing this bug results in much more stable arm movement.
2. The head camera offset was too small, so that when the robot moved around, the head camera would be occluded by the head geometry.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by aclegg3 — COMMENTED** [review]
These changes make sense. 

Looks like the gripper open test is failing, maybe because it used to have arm motor settings and now has default settings?

File: src_python/habitat_sim/utils/datasets_download.py [hunk 1]
```diff
@@ -109,16 +109,16 @@ def initialize_test_data_sources(data_path):
...
+            "version": "1.4",
         },
         "replica_cad_baked_lighting": {
-            "source": "https://dl.fbaipublicfiles.com/habitat/ReplicaCAD/ReplicaCAD_baked_lighting_v1.3.zip",
-            "package_name": "ReplicaCAD_baked_lighting_v1.3.zip",
+            "source": "https://dl.fbaipublicfiles.com/habitat/ReplicaCAD/ReplicaCAD_baked_lighting_v1.4.zip",
+            "package_name": "ReplicaCAD_baked_lighting_v1.4.zip",
             "link": data_path + "replica_cad_baked_lighting",
-            "version": "1.3",
+            "version": "1.4",
```
Comment (aclegg3) [inline src_python/habitat_sim/utils/datasets_download.py:121]: Let's keep these changes isolated to #1694, so remove before merge.

Comment (ASzot) [reply src_python/habitat_sim/utils/datasets_download.py]: Done

**Review by aclegg3 — APPROVED** [review]
LGTM. When tests pass, let's merge.

**Pull request merged**